### ISSUE TITLE
SwallowWebExceptions is being changed to true

### ISSUE THREAD

**Issue by CYBERRRR** [linvi/tweetinvi#473]
Hi (great job on this project!)
I have a Twitter service class that sets `ExceptionHandler.SwallowWebExceptions = false;` in its constructor.

One of the methods has the following:
`var userCredentials = new TwitterCredentials(Credentials.ConsumerKey, Credentials.ConsumerSecret, accessToken, accessSecret);`

...

`var result = await Auth.ExecuteOperationWithCredentials(userCredentials, () => TwitterAccessorAsync.ExecutePOSTQuery<IUserDTO>(string.Format(USERS_SHOW, handle)));
`

**Before** the call to Auth.ExecuteOperationWithCredentials() I can confirm that SwallowWebExceptions is set to **false** but immediately after executing it is already set to **true** and the surrounding generic try catch doesn't catch anything. (I know the method should be a GET, this is just so I can easily force an exception).

Checking the last exception from ExceptionHandler has the exception I want to catch.
I've read around the wiki but I can't seem to find what I am doing wrong.
Any help would be appreciated, thanks in advance!

**Comment by JoshKeegan** [comment]
I think the exception handler is thread-specific.
To rule out any problems being caused by that, have you tried using the non-async methods?
Something like (untested):
```cs
var result = Auth.ExecuteOperationWithCredentials(userCredentials, () => TwitterAccessor.ExecutePOSTQuery<IUserDTO>(string.Format(USERS_SHOW, handle)));
```
Cheers,
Josh

**Comment by linvi** [comment]
Thanks for reporting this issue. It looks like a bug that I will fix in the coming release.

**Comment by CYBERRRR** [comment]
Thank you for your help linvi and Josh!

**Comment by linvi** [comment]
I have made some progress on the issue. It is a very important problem that is resulting from the `SynchronizationContext` of some type of applications.

Could you please let me know what type of application you are using.

I can personally reproduce the problem on Console applications.

**Comment by CYBERRRR** [comment]
Sure, it's a .NET Core MVC web app running on Ubuntu.

Em 12/05/2017 2:52 da tarde, linvi <notifications@github.com> escreveu:

I have made some progress on the issue. It is a very important problem that is resulting from the SynchronizationContext of some type of applications.

Could you please let me know what type of application you are using.

I can personally reproduce the problem on Console applications.

**Comment by linvi** [comment]
The problem is a critical problem of how the library has been used.
The second problem is that I cannot fix this issue for Portable Class Libraries as they do not allow developers to change the `SynchronizationContext`.

As a result this fix will be part of a new `Tweetinvi 2.0` which will be entirely dedicated to this specific problem.

In the meantime please do the following:

* Install https://www.nuget.org/packages/Nito.AsyncEx

Slightly update your code to : 

``` c#
var result = AsyncContext.Run(Auth.ExecuteOperationWithCredentials(userCredentials, 
    () => TwitterAccessorAsync.ExecutePOSTQuery<IUserDTO>(string.Format(USERS_SHOW, handle)));
);
```

Thank you for your time and patience.

**Comment by CYBERRRR** [comment]
Thank you for trying linvi. I will try out your recommended workaround, hope it works ;)

Em 21/05/2017 5:54 da tarde, linvi <notifications@github.com> escreveu:

The problem is a critical problem of how the library has been used.
The second problem is that I cannot fix this issue for Portable Class Libraries as they do not allow developers to change the SynchronizationContext.

As a result this fix will be part of a new Tweetinvi 2.0 which will be entirely dedicated to this specific problem.

In the meantime please do the following:

  *   Install https://www.nuget.org/packages/Nito.AsyncEx

Slightly update your code to :

var result = AsyncContext.Run(Auth.ExecuteOperationWithCredentials(userCredentials,
    () => TwitterAccessorAsync.ExecutePOSTQuery<IUserDTO>(string.Format(USERS_SHOW, handle)));
);

Thank you for your time and patience.

**Comment by linvi** [comment]
Verified on all platforms :)
